poutyne/framework/model.py

```diff
@@ -136,18 +136,30 @@ class Model:
 
     """
 
-    def __init__(self, model, optimizer, loss_function, *, metrics=None, batch_metrics=None, epoch_metrics=None):
+    def __init__(self,
+                 network,
+                 optimizer,
+                 loss_function,
+                 *,
+                 metrics=None,
+                 batch_metrics=None,
+                 epoch_metrics=None,
+                 model=None):
```

Comment:
With this current implementation, I can pass a model arg to my Model initialization, but it is undocumented and does absolutely nothing since it is not even assigned to self.network. I would definitely keep it in the args right now, if there is a model arg, first output a warning of future deprecation, then if there is also a non-null network, raise the invalid config error.

Reply:
1. `model = Model(pytorch_network, optimizer, ...)` then the pytorch_network will be pass to network. 

2.1 `model = Model(something_else, model=pytorch_network, optimizer = optimizer, ...)` depending of the case the something_else will be set to network, so the `if network and model is not None:` should catch this case.

2.2  `model = Model(model=pytorch_network, optimizer = optimizer, ...)`  will fail since the constructor is not correctly set. So Python will raise the appropriated error. (need to change is code)

I think we got all the edge case, but you are right, I forgot to add depreciation warning in the doc.

Reply:
added doc in ebc7fc2

Reply:
In that case, since 2.1 and 2.2 crashes anyway, I don't see why we need the model argument. It won't be more backward compatible with or without it.

Reply:
True. So I just raise a depreciation warning in all case and remove the `if network and model is not None:` test ?

Reply:
I don't see where you could raise a deprecation warning. Just remove the `if network and model is not None:` test.

Reply:
Oh if you meant the deprecation warning in the model property, yeah, sure, keep that one.

Reply:
solved in d65545a